Tools/CMake/AMReXParallelBackends.cmake

```diff
@@ -284,7 +284,11 @@ if (AMReX_HIP)
 
    # ROCm 5.5: hipcc now relies on clang to offload code objects from (.a) archive files,
    # so we need to tell the offload-linker to include all code objects in archives.
-   target_link_options(amrex PUBLIC -Xoffload-linker --whole-archive)
+   include(CheckLinkerFlag)
+   check_linker_flag("-Xoffload-linker --whole-archive" LINKER_HAS_WHOLE_ARCHIVE_OFFLOAD)
```

Comment:
Missing the first argument for <lang>?

Reply:
Also this is a new feature introduced in 3.18.  We would have to bump the minimum requirement from 3.17 to 3.18.

Reply:
oh, yes good point!